test/form_builder/general_test.rb

```diff
@@ -117,8 +117,19 @@ def with_custom_form_for(object, *args, &block)
     assert_select 'form input#user_name.string'
   end
 
-  test 'builder generates text areas for text columns' do
+  test 'builder generates text field for text columns' do
     with_form_for @user, :description
+    if @user.respond_to?(:type_for_attribute) &&
+       @user.type_for_attribute('description').type == :string
+      assert_select 'form input#user_description.string'
```

Comment:
@rafaelfranca I think that this means that database-specific types rely on the cooperation of the database adapter to distinguish between `:text` and `:string` as they are both represented in Ruby by `String`. As the testing environment does not include any database adapter, it must rely on the mocking of some of the type information where ActiveRecord might differ from ActiveModel.

Supporting the current range of Rails (actually, ActiveRecord and ActiveModel) versions means that things are a bit fiddly now with the `:text` type and there might need to be guidance on manually indicating inputs which should be rendered with a `textarea` in projects that can no longer automatically distinguish `:string` from `:text`